### PR TITLE
add module configuration for demo handler

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <oliver.lamy@gmail.com>

### CODE REVIEW

**Comment by olamy** [comment]
finally maybe a name such `demo-core-handler` would be better. then we could have more `demo-core-xxx` 
@sbordet @joakime @gregw ?

**Comment by sbordet** [comment]
I prefer just `demo-handler` or `core-demo-handler` since the environment is always first.

File: jetty-core/jetty-demos/jetty-demo-handler/src/main/config/modules/demo-handler.mod [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# DO NOT EDIT - See: https://www.eclipse.org/jetty/documentation/current/startup-modules.html
+
+[description]
+Demo Core Handler
```
Comment (sbordet) [inline jetty-core/jetty-demos/jetty-demo-handler/src/main/config/modules/demo-handler.mod]: Remove "Core"

File: jetty-core/jetty-demos/jetty-demo-handler/src/main/config/modules/demo-handler.mod [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+[tags]
+demo
+handler
+
+[depends]
+server
+
+[files]
+basehome:modules/demo.d/demo-handler.xml|etc/demo-handler.xml
+maven://org.eclipse.jetty.demos/jetty-demo-handler/${jetty.version}/jar|lib/jetty-demo-handler.jar
```
Comment (sbordet) [inline jetty-core/jetty-demos/jetty-demo-handler/src/main/config/modules/demo-handler.mod:15]: Is this correct that it ends in `/jar`, before the `|`?

File: jetty-core/jetty-demos/jetty-demo-handler/src/main/config/modules/demo.d/demo-handler.xml [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
-  </Set>
-</Configure>
+    </Set>
+    <Set name="contextPath">/demo-handler</Set>
+  </New>
+
+  <Call name="addHandler">
+    <Arg><Ref refid="helloHandler"/></Arg>
+  </Call>
+</Configure>
```
Comment (sbordet) [inline jetty-core/jetty-demos/jetty-demo-handler/src/main/config/modules/demo.d/demo-handler.xml]: This should really be:

```xml
<Configure id="Contexts" class="org.eclipse.jetty.server.handler.ContextHandlerCollection">
  <Call name="addHandler">
    <Arg>
      <New id="helloHandler" class="org.eclipse.jetty.server.handler.ContextHandler">
        <Set name="handler">
          <New class="org.eclipse.jetty.demo.HelloHandler" />
        </Set>
        <Set name="contextPath">/demo-handler</Set>
      </New>
    </Arg>
  </Call>
</Configure>
```

File: tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/DemoModulesTests.java [hunk 4]
```diff
@@ -421,4 +422,49 @@ public void testSessionDump(String env) throws Exception
...
+        String jettyVersion = System.getProperty("jettyVersion");
+        JettyHomeTester distribution = JettyHomeTester.Builder.newInstance()
+                .jettyVersion(jettyVersion)
+                .jettyBase(jettyBase)
+                .mavenLocalRepository(System.getProperty("mavenRepoPath"))
+                .build();
+
+        int httpPort = distribution.freePort();
+        int httpsPort = distribution.freePort();
+        assertThat("httpPort != httpsPort", httpPort, is(not(httpsPort)));
```
Comment (sbordet) [inline tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/DemoModulesTests.java]: You are not using the `https` module, so no need to allocate an https port.

File: tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/DemoModulesTests.java [hunk 5]
```diff
@@ -421,4 +422,49 @@ public void testSessionDump(String env) throws Exception
...
+
+        try (JettyHomeTester.Run runConfig = distribution.start(argsConfig))
+        {
+            assertTrue(runConfig.awaitFor(START_TIMEOUT, TimeUnit.SECONDS));
+            assertEquals(0, runConfig.getExitValue());
+
+            String[] argsStart = {
+                    "jetty.http.port=" + httpPort,
+                    "jetty.httpConfig.port=" + httpsPort,
+                    "jetty.ssl.port=" + httpsPort
```
Comment (sbordet) [inline tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/DemoModulesTests.java]: These 2 lines are unnecessary.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-core/jetty-demos/jetty-demo-handler/src/main/config/modules/demo-handler.mod (hunk 2 above)
Comment (olamy) [reply jetty-core/jetty-demos/jetty-demo-handler/src/main/config/modules/demo-handler.mod:15]: yes. the format is `maven://<groupId>/<artifactId>/<version>/<type>/<classifier>`
with `type` optional default to `jar`

**Review by sbordet — APPROVED** [review]

**Pull request merged**